test/units/testsuite-48.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -ex
+
+test_rule="/run/udev/rules.d/48-test.rules"
+
+setup() {
+    mkdir -p "${test_rule%/*}"
+    cp -f /etc/udev/udev.conf /etc/udev/udev.conf.bckp
+    echo 'KERNEL=="lo", SUBSYSTEM=="net", PROGRAM="/usr/bin/sleep 2"' > "${test_rule}"
+    echo "event_timeout=1" >> /etc/udev/udev.conf
+    echo "timeout_signal=SIGABRT" >> /etc/udev/udev.conf
+
+    systemctl restart systemd-udevd.service
+}
+
+teardown() {
+    set +e
+
+    mv -f /etc/udev/udev.conf.bckp /etc/udev/udev.conf
+    rm -f "$test_rule"
+    systemctl restart systemd-udevd.service
+}
+
+run_test() {
+    since="$(date +%T)"
+
+    echo add > /sys/class/net/lo/uevent
+    sleep 5
+
+    coredumpctl --since "$since" --until="$(date +%T)" --no-legend --no-pager | grep /usr/bin/sleep
```

Comment:
The tests checks for coredump generated for `/usr/bin/sleep` but it's actually `/usr/lib/systemd/systemd-udevd` who SIGABRTs:

```
[   16.867722] systemd[1]: Received SIGCHLD from PID 349 (sleep).
[   16.868117] systemd[1]: Child 349 (sleep) died (code=exited, status=0/SUCCESS)
[   16.868206] systemd-udevd[336]: Worker [339] terminated by signal 6 (ABRT)
[   16.868630] systemd-udevd[336]: lo: Worker [339] failed
[   16.868844] systemd[1]: systemd-udevd.service: Child 349 belongs to systemd-udevd.service.
[   16.947668] systemd-coredump[351]: Process 339 (systemd-udevd) of user 0 dumped core.
                                      
                                      Stack trace of thread 339:
                                      #0  0x00007f9ed40376ca epoll_wait (libc.so.6 + 0xff6ca)
                                      #1  0x00007f9ed4368010 sd_event_wait (libsystemd-shared-245.so + 0x1b0010)
                                      #2  0x00007f9ed4368927 sd_event_run (libsystemd-shared-245.so + 0x1b0927)
                                      #3  0x00007f9ed43689ca sd_event_loop (libsystemd-shared-245.so + 0x1b09ca)
                                      #4  0x00005620ce75d120 n/a (systemd-udevd + 0x1a120)
                                      #5  0x00005620ce75ef9e n/a (systemd-udevd + 0x1bf9e)
                                      #6  0x00005620ce76a858 n/a (systemd-udevd + 0x27858)
                                      #7  0x00005620ce76e891 n/a (systemd-udevd + 0x2b891)
                                      #8  0x00005620ce76ec99 n/a (systemd-udevd + 0x2bc99)
                                      #9  0x00005620ce75f11f n/a (systemd-udevd + 0x1c11f)
                                      #10 0x00005620ce7589b9 n/a (systemd-udevd + 0x159b9)
                                      #11 0x00005620ce758d2e n/a (systemd-udevd + 0x15d2e)
                                      #12 0x00007f9ed433d130 n/a (libsystemd-shared-245.so + 0x185130)
                                      #13 0x00007f9ed43669c2 n/a (libsystemd-shared-245.so + 0x1ae9c2)
                                      #14 0x00007f9ed43686ef sd_event_dispatch (libsystemd-shared-245.so + 0x1b06ef)
                                      #15 0x00007f9ed436894d sd_event_run (libsystemd-shared-245.so + 0x1b094d)
                                      #16 0x00007f9ed43689ca sd_event_loop (libsystemd-shared-245.so + 0x1b09ca)
                                      #17 0x00005620ce759062 n/a (systemd-udevd + 0x16062)
                                      #18 0x00005620ce759601 n/a (systemd-udevd + 0x16601)
                                      #19 0x00005620ce75998c n/a (systemd-udevd + 0x1698c)
                                      #20 0x00005620ce759b69 n/a (systemd-udevd + 0x16b69)
                                      #21 0x00005620ce759efb n/a (systemd-udevd + 0x16efb)
                                      #22 0x00007f9ed433d130 n/a (libsystemd-shared-245.so + 0x185130)
                                      #23 0x00007f9ed43669c2 n/a (libsystemd-shared-245.so + 0x1ae9c2)
                                      #24 0x00007f9ed43686ef sd_event_dispatch (libsystemd-shared-245.so + 0x1b06ef)
                                      #25 0x00007f9ed436894d sd_event_run (libsystemd-shared-245.so + 0x1b094d)
                                      #26 0x00007f9ed43689ca sd_event_loop (libsystemd-shared-245.so + 0x1b09ca)
                                      #27 0x00005620ce756fc9 n/a (systemd-udevd + 0x13fc9)
                                      #28 0x00005620ce758204 n/a (systemd-udevd + 0x15204)
                                      #29 0x00005620ce75afa7 n/a (systemd-udevd + 0x17fa7)
                                      #30 0x00007f9ed3f5f023 __libc_start_main (libc.so.6 + 0x27023)
                                      #31 0x00005620ce754bde n/a (systemd-udevd + 0x11bde)
```

coredumpctl output: https://ci.centos.org/job/systemd-pr-build-vagrant/6614/artifact//systemd-centos-ci/artifacts_zbQOFb/vagrant-logs.e3J/vagrant-arch-testsuite.Khb/TEST-48-UDEV-EVENT-TIMEOUT_coredumpctl_collect_FAIL.log

Reply:
That is expected. Both worker and helper process get killed by the same signal.

@mrc0mmand are you sure it was only `systemd-udevd` that aborted?

Reply:
Yep, only `systemd-udev` died with `ABRT` in both cases:

```
Apr 17 14:54:18 systemd-testsuite systemd[496]: systemd-coredump@0-495-0.service: Executing: /usr/lib/systemd/systemd-coredump
Apr 17 14:54:18 systemd-testsuite systemd-udevd[487]: Worker [491] terminated by signal 6 (ABRT)
Apr 17 14:54:18 systemd-testsuite systemd-udevd[487]: lo: Worker [491] failed
Apr 17 14:54:18 systemd-testsuite systemd[1]: Received SIGCHLD from PID 492 (sleep).
Apr 17 14:54:18 systemd-testsuite systemd[1]: Child 492 (sleep) died (code=killed, status=15/TERM)
Apr 17 14:54:18 systemd-testsuite systemd-coredump[496]: Failed to generate stack trace: Unwinding not supported for this architecture
Apr 17 14:54:18 systemd-testsuite systemd-coredump[496]: Process 491 (systemd-udevd) of user 0 dumped core.
```